client/my-sites/checkout/checkout/web-payment-box.jsx

```diff
@@ -28,523 +26,373 @@ import {
+	onSubmit: PropTypes.func.isRequired,
+	presaleChatAvailable: PropTypes.bool,
+	translate: PropTypes.func.isRequired,
+	disablePostalCodeDebounce: PropTypes.bool,
+};
 
-	/**
-	 * @return {PaymentRequest} A configured payment request object.
-	 */
-	getPaymentRequestForBasicCard = () => {
-		const { cart } = this.props;
+function WebPayButton( {
+	stripe,
+	isStripeLoading,
+	stripeConfiguration,
+	countryCode,
+	postalCode,
+	cart,
+	onSubmit,
+	paymentType,
+	translate,
+} ) {
+	const { currency, total_cost_integer, sub_total_integer, total_tax_integer } = cart;
+	const isRenewal = hasRenewalItem( cart );
+	const shouldDisplayItems = shouldShowTax( cart );
+	useEffect( () => {
+		stripe && setStripeObject( stripe, stripeConfiguration );
+	}, [ stripe, stripeConfiguration ] );
+
+	const paymentRequestOptions = usePaymentRequestOptions( {
+		translate,
+		sub_total_integer,
+		total_tax_integer,
+		total_cost_integer,
+		countryCode: getProcessorCountryFromStripeConfiguration( stripeConfiguration ),
+		currency,
+		shouldDisplayItems,
+	} );
+	const { paymentRequest, canMakePayment } = useStripePaymentRequest( {
+		stripe,
+		paymentRequestOptions,
+		countryCode,
+		postalCode,
+		onSubmit,
+		isRenewal,
+	} );
+
+	if ( ! stripe && ! isStripeLoading ) {
+		throw new Error( 'Stripe is required but not available' );
+	}
+	const disabledReason = getDisabledReason( {
+		isStripeLoading,
+		canMakePayment,
+		postalCode,
+		countryCode,
+		translate,
+	} );
+	return (
+		<PaymentRequestButton
+			paymentRequest={ paymentRequest }
+			isRenewal={ isRenewal }
+			paymentType={ paymentType }
+			translate={ translate }
+			disabled={ !! disabledReason }
+			disabledReason={ disabledReason }
+		/>
+	);
+}
 
-		const supportedPaymentMethods = [
-			{
-				supportedMethods: WEB_PAYMENT_BASIC_CARD_METHOD,
-				data: {
-					supportedNetworks: SUPPORTED_NETWORKS,
-				},
-			},
-		];
+WebPayButton.propTypes = {
+	stripe: PropTypes.object,
+	countryCode: PropTypes.string,
+	postalCode: PropTypes.string,
+	cart: PropTypes.shape( {
+		currency: PropTypes.string.isRequired,
+		total_cost: PropTypes.number.isRequired,
+		products: PropTypes.array.isRequired,
+	} ).isRequired,
+	onSubmit: PropTypes.func.isRequired,
+	translate: PropTypes.func.isRequired,
+};
 
-		const paymentDetails = this.getPaymentDetails( cart );
+function getProcessorCountryFromStripeConfiguration( stripeConfiguration ) {
+	return stripeConfiguration.processor_id === 'stripe_ie' ? 'IE' : 'US';
+}
 
-		return new PaymentRequest( supportedPaymentMethods, paymentDetails, PAYMENT_REQUEST_OPTIONS );
-	};
+function getDisabledReason( {
+	isStripeLoading,
+	canMakePayment,
+	postalCode,
+	countryCode,
+	translate,
+} ) {
+	if ( isStripeLoading ) {
+		return translate( 'Loading…' );
+	}
+	if ( ! canMakePayment ) {
+		return translate( 'This payment method is unavailable' );
+	}
+	if ( ! postalCode || ! countryCode ) {
+		return translate( 'Please specify a country and postal code' );
```

Comment:
Hi! I've found a possible matching string that has already been translated **31** times:
`translate( 'Please specify a country and postal code.' )` *ES Score: 9*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

Reply:
Correct, but the way this is formatted I think it would be better without the period.

Reply:
I've fixed the other occurrence, so we can still end up with just one string versions (the translations should be copied over)